handler/OpenIDLoginHandler.php

```diff
@@ -0,0 +1,281 @@
+        }
+
+        $request->redirect(Application::get()->getRequest()->getContext()->getPath(), 'index');
+
+        return false;
+    }
+
+    /**
+     * Used for legacy login in case of errors or other bad things.
+     */
+    function legacyLogin(array $args, Request $request)
+    {
+        $templateMgr = TemplateManager::getManager($request);
+        $this->_enableLegacyLogin($templateMgr, $request);
+        $templateMgr->assign('disableUserReg', true);
+
+        return $templateMgr->display('frontend/pages/userLogin.tpl');
+    }
+
+    /**
+     * Overwrites the default registration, because it is not needed anymore.
+     * User registration is done via OpenID provider.
+     */
+    function register(array $args, Request $request)
+    {
+        $this->index($args, $request);
+    }
+
+    /**
+     * Overwrites default signOut.
+     * Performs OJS logout and if logoutUrl is provided (e.g. Apple doesn't provide this url) it redirects to the oauth logout to delete session and tokens.
+     */
+    function signOut(array $args, Request $request)
+    {
+        if (Validation::isLoggedIn()) {
+            $router = $request->getRouter();
+
+            $requestUser = $request->getUser();
+
+            $user = null;
+            $lastProvider = null;
+            if ($requestUser) {
+                $user = Repo::user()->get($requestUser->getId());
+
+                $lastProvider = $user->getData('openid::lastProvider');
+            }
+            
+            $context = Application::get()->getRequest()->getContext();
+            $contextId = ($context == null) ? PKPApplication::CONTEXT_SITE : $context->getId();
+            $settingsJson = $this->plugin->getSetting($contextId, 'openIDSettings');
+
+            if (isset($user)) {
+                $user->setData('openid::lastProvider', null);
+                Repo::user()->edit($user);
+            }
+
+            $sessionManager = SessionManager::getManager();
+            $session = $sessionManager->getUserSession();
+            $token = $session->getSessionVar('token');
+
+            Validation::logout();
+            if (isset($settingsJson, $lastProvider)) {
+                $providerList = json_decode($settingsJson, true)['provider'];
+                $settings = $providerList[$lastProvider];
+
+                if (!empty($settings['logoutUrl']) && isset($settings['clientId'])) {
+                    $request->redirectUrl(
+                        $settings['logoutUrl'] .
+                        '?client_id=' . urlencode($settings['clientId']) .
+                        '&post_redirect_uri=' . urlencode($router->url($request, $context->getPath(), "index")) .
+                        '&id_token_hint=' . urlencode($token)
+                    );
+                }
+            }
+        }
+        $request->redirect(Application::get()->getRequest()->getContext()->getPath(), 'index');
```

Comment:
Could just be `$request->redirect(null, 'index');`.

Reply:
Removed